### DISCUSSION TITLE
Missing MMFF parameters for the terminal oxygen in RO[O] molecules?

### DISCUSSION THREAD

**Discussion by xiaoruiDong** [rdkit/rdkit#3791] [Q&A]
Hi all,

In my research work (combustion chemistry), ROO molecules (molecules with -O-O. groups) are very important. An example is shown below:

**SMILES: CCO[O]**
![image](https://user-images.githubusercontent.com/35771233/107137085-9a86a880-68ce-11eb-960d-fc997f998522.png)

Here, I want to use the MMFF94s force field to optimize those molecules, but it seems like there is no MMFF parameter for the terminal oxygen in the RDKit. 

```
mol = AllChem.MolFromSmiles('CCO[O]')
mol = AllChem.AddHs(mol)
AllChem.MMFFGetMoleculeProperties(mol, mmffVariant='mmff94s')
>>> None  # If input 'CCOO' rather than CCO[O], parameter object will be returned
```

Thus can not be optimized @ MMFF94s with RDKit. However, this molecule can be optimized at MMFF94s in the Avogadro software (using Openbabel as the backend optimizer, I think).

I am not sure about the differences in implementations between RDKit and Openbabel, but I suppose that both packages should behave similarly. Do you know, is that a bug in the RDKit? or parameters are actually missing in the RDKit?

**Comment by edwardclem** [comment]
Hi Xiaorui, there are a few groups that are parametrized in openbabel but not in RDKit's implementation of the same force field (-SF5 is one, I believe), so I don't think that's necessarily an RDKit bug. My guess is that the RDKit MMFF implementation probably just doesn't have a parameter for the RO[O] radical.

**Reply by xiaoruiDong** [reply]
Thanks a lot, @edwardclem! I also find some other CHON groups have no corresponding parameters. I am using RDKit's python API, do you know if there is a way to add parameters for such groups?

**Reply by He-Ziping** [reply]
Hello, @edwardclem!  I'm still new to this area and trying to understand how parametrization works. You mentioned that -SF₅ is parametrized in Open Babel — may I ask how you obtained those parameters or where I could find them? I tried to get them myself but couldn’t succeed. I'd really appreciate your help!

**Comment by ptosco** [comment]
@xiaoruiDong native MMFF94 has no parameters for radicals. You may check the list of parameters here:

https://github.com/openbabel/openbabel/blob/master/data/mmffdef.par

So your radical cannot be minimized as such with MMFF94.

**Reply by xiaoruiDong** [reply]
Many thanks, @ptosco! It is really nice of you to let me aware of this reference. I didn't realize that since other radicals seem to work well (e.g., C[CH2], CC[O], etc.)... However, I still need RDKit to do large scale MMFF94 conformer optimization. I am currently using a naive workaround to treat this type of molecule (RO[O]): 

1. Before optimization, I manually replace the terminal oxygen with a fluorine atom (presumably, they behave similarly). 
2. The artificial molecule can be optimized by RDKit MMFF94s, and I optimize this molecule.
3. After optimization, I change the fluorine atom back to an oxygen atom.

It is no different from using terminal F's parameter as RO[O]'s parameter. I know this seems non-desirable... 
In general, can I manually add parameters for such a group via Python API? Is there a smarter way to do something like this? I really appreciate your help!

**Reply by xiaoruiDong** [reply]
Another quick question, RDKit force field (MMFF94) cannot deal with intermolecular interactions, is that correct? I put something like 'O=C.O=C' into RDKit, and I found that molecule conformation was optimized but the distance between molecules was not.

**Reply by ptosco** [reply]
As mentioned in the docs, intermolecular interactions are switched off by default, but they can be turned on:

https://www.rdkit.org/docs/source/rdkit.Chem.rdForceFieldHelpers.html?highlight=mmffget#rdkit.Chem.rdForceFieldHelpers.MMFFGetMoleculeForceField

if you wish to model a radical you should probably use QM rather than MM. Adding force field parameters is possible from C++, not from Python.

**Reply by xiaoruiDong** [reply]
Thanks for letting me know about this argument! ... I can't believe I missed that one. Regarding radicals, yes, I actually do many QM calculations. I am using RDKit to generate many conformations and use them as initial guesses for QM. However, the function of embed molecule / multiple conformers doesn't fulfill my need for this. Thanks for the advice anyway!
